src/Libraries/PythonNodeModelsWpf/ScriptEditorWindow.xaml.cs

```diff
@@ -267,6 +266,7 @@ private void OnEngineChanged(object sender, System.Windows.Controls.SelectionCha
 
         private void OnScriptEditorWindowClosed(object sender, EventArgs e)
         {
+            completionProvider?.Dispose();
```

Comment:
I know I suggested this - have you had time to test that a new completion provider is created when the script editor window is reopened?

Reply:
YEs, it is